inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -515,6 +541,29 @@ public void writeMessage(InputStream message) {
           statsTraceCtx.outboundMessageSent(outboundSeqNo, -1, -1);
           clientStream.statsTraceCtx.inboundMessage(outboundSeqNo);
           clientStream.statsTraceCtx.inboundMessageRead(outboundSeqNo, -1, -1);
+
+          long messageLength;
+          try {
+            if (assumedMessageSize != -1) {
+              messageLength = assumedMessageSize;
+            } else if (message instanceof KnownLength || message instanceof ByteArrayInputStream) {
+              messageLength = getKnownLength(message);
```

Comment:
Just call `inputStream.available()` directly here? This is already within a try block that catches IOException, and there's no reason for one to be StatusRuntimeException and the other to be regular RuntimeException.

Reply:
Alright sounds good.